vendor/bat-native-ledger/src/bat_client.cc

```diff
@@ -534,4 +546,37 @@ void BatClient::getGrantCaptchaCallback(bool success,
   ledger_->OnGrantCaptcha(response, it->second);
 }
 
+void BatClient::getGrantViaSafetynetCheck() {
+  auto request_id =
+      ledger_->LoadURL(braveledger_bat_helper::buildURL(
+          (std::string)GET_PROMOTION_ATTESTATION + ledger_->GetPaymentId(),
+          PREFIX_V1),
+      std::vector<std::string>(), "", "",
+      ledger::URL_METHOD::GET, &handler_);
+  handler_.AddRequestHandler(std::move(request_id),
+                      std::bind(&BatClient::getGrantViaSafetynetCheckCallback,
+                                       this,
+                                       _1,
+                                       _2,
+                                       _3));
+}
+
+void BatClient::getGrantViaSafetynetCheckCallback(bool success,
+                                                  const std::string& response,
+                                                  const std::map<std::string,
+                                                  std::string>& headers) {
+  ledger_->LogResponse(__func__, success, response, headers);
+
+  if (!success) {
+    // Attestation failed
+    braveledger_bat_helper::GRANT grant;
+    ledger_->OnGrantFinish(ledger::Result::SAFETYNET_ATTESTATION_FAILED,
+        grant);
+    return;
+  }
+  std::string nonce;
+  braveledger_bat_helper::getJSONValue("nonce", response, nonce);
+  ledger_->OnGrantViaSafetynetCheck(nonce);
```

Comment:
how is this nonce generated?

Reply:
We receive it from server. I believe @evq knows how it is generated on server side.